internal/parse/parser.go

```diff
@@ -273,11 +302,152 @@ func (p *Parser) parseGoFullName() (FullName, bool, error) {
 		return FullName{}, false,
 			fmt.Errorf("go object near char %d not qualified", p.pos)
 	}
+
 	cp.restore()
 	return FullName{}, false, nil
 }
 
-// parseInputExpression parses an input expression of the form $Type.name.
+// parseList takes a parsing function that returns a FullName and parses a
+// bracketed, comma seperated, list.
+func (p *Parser) parseList(parseFn func(p *Parser) (FullName, bool, error)) ([]FullName, bool, error) {
+	cp := p.save()
+
+	if p.skipByte('(') {
+		parenPos := p.pos
+		p.skipSpaces()
+
+		nextItem := true
+		var objs []FullName
+		for nextItem {
+			if obj, ok, err := parseFn(p); ok {
+				objs = append(objs, obj)
+			} else if err != nil {
+				return nil, false, err
+			} else {
+				return nil, false, fmt.Errorf("invalid identifier near char %d", p.pos)
```

Comment:
The way it currently works is that the caller of parseList decides weather to heed the error or not. If they are trying to parse something they know is an expression they can pass the error on, otherwise they will ignore it.

Changed the comment.

Reply:
With the new logic for parsing `(&P.a, &P.b)` this did need changing. It now works as follows:
- If the first item in the list doesn't parsed we return `ok=false, err=nil`
- If the first item did parse and any subsequent items dont/the closing bracket is missing we throw an error

Reply:
Nice, that sounds like helpful behavior.